### DISCUSSION TITLE
K1C printing with no filament installed after helper_script.sh

### DISCUSSION THREAD

**Discussion by smckown** [Guilouz/Creality-Helper-Script-Wiki#715] [Issues]
I installed fluidd, KAMP, and a few other things.  I did not install M600, guppy screen, nor did I remove the CP web service (port 80).

After the helper script upgrade, with no filament loaded, the printer reports filament runout, as visible from port 80 (CP), port 4408 (fluidd), and also the printer LCD screen.  Yet the printer will print anyway.  It doesn't matter if I upload a new file from Orca or if I select from one of the UI's a previously printed file.  I let the printer print each  job I tried until it started on the first layer.

I started from Creality firmware 1.3.3.36, which came pre-installed on the printer.  I did not do a factory reset before running the script, as I only set up WiFi and printed about 2 dozen times to wring out the hardware prior to running the helper script.

Similar discussions here seem quite different than what I'm experiencing.

Any suggestions on what the problem could be, or some pointers to troubleshooting?

**Comment by kadabriel** [comment]
I also noticed this on the k1SE, but i figure it is not an issue. so you can make a top mount for TPU and buypass it with out jaming a filament in the sensor to trick it.
I am jus curious to see if the print pauses if the sensored is triggered during a print. then it would be perfect.

**Comment by smckown** [comment]
> I am jus curious to see if the print pauses if the sensored is triggered during a print. then it would be perfect.

For me, the standard runout sensor does not cause the print to pause if it triggers.  I tested using the technique @kadabriel referenced: filament loaded directly to the extruder and a piece of filament pushed into the runout sensor.  Once a print was in progress, I pulled the filament piece out of the sensor.  fluidd indicated the sensor had been triggered but the printer did not pause printing.

I didn't test runout before using the Helper Script, so am unsure if it is broken in the Creality default firmware or the Helper Script broke something.  I didn't make any manual tweaks to printer.cfg before the runout sensor tests after running the Helper Script.  Some day I'll go back and look at this, as I'd like for it to work correctly.

Since then, I've added a runout sensor in the extruder.  With a configuration stanza for it in printer.cfg and printing something with just a short piece of filament loaded in the extruder, the print will pause when the toolhead runout sensor triggers.  Then I can use the touch panel to load new filament and continue the print without incident.

PS - the toolhead runout sensor is documented here: via https://www.printables.com/model/651002-k1-extruder-filament-runout-sensor.

**Reply by kadabriel** [reply]
Tnx, i will do what you did!

**Comment by kadabriel** [comment]
I am sure I did not do anything to the original firmware. I bet we are not the only ones experiencing this.
 the sensor is working, it gets triggered, only the problem is that it does not pause or stop the print. should have a script that triggers M600 when filament sensor is triggered.

**Comment by kadabriel** [comment]
@smckown, I just tested this out. I cut, the filament below the sensor when I started the print. when the sensor got triggered it paused the print automatically.
I realised that if you have Home Assistant and the Moonraker integration. I can automate the M600 to trigger when the sensor does not have filament.  Are you sure you put The fillament all the way in, that it got the blue light on? if i start a print without filament the printer just prints air but i get a message in fluid like you describe,

**Comment by smckown** [comment]
Aha!  The rear runout sensor is working for me after all, I just wasn't patient enough.  I loaded filament but cut it with only 2" sticking out the input side of the rear runout sensor, then started a print.  fluidd showed the runout sensor triggering immediately when the end of filament passed through it, but the printer continued to print for a a good bit, using up another 4" of filament before the pause occurred.  Thanks @kadabriel for encouraging me to look at this again!
